platform/ios/src/MGLMapView.mm

```diff
@@ -390,6 +393,8 @@ - (void)commonInit
 {
     MGLinitializeRunLoop();
 
+    _isWaitingForStyleLoad = YES;
```

Comment:
Setting or reloading the style URL should also cause us to wait for the style to load. If you want to be sure the flag gets set at the right times, replace any occurrences of `_style` with `self.style`, then override `-setStyle:` to set this flag.